packages/adapter-drizzle/src/lib/pg.ts

```diff
@@ -19,83 +19,100 @@ import type {
+        token_type: text("token_type"),
+        scope: text("scope"),
+        id_token: text("id_token"),
+        session_state: text("session_state"),
+      },
+      (table) => {
+        return {
+          compositePk: primaryKey({
+            columns: [table.provider, table.providerAccountId],
+          }),
+        }
+      }
+    ) satisfies DefaultPostgresAccountsTable)
 
-export const postgresSessionsTable = pgTable("session", {
-  sessionToken: text("sessionToken").primaryKey(),
-  userId: text("userId")
-    .notNull()
-    .references(() => postgresUsersTable.id, { onDelete: "cascade" }),
-  expires: timestamp("expires", { mode: "date" }).notNull(),
-}) satisfies DefaultPostgresSessionsTable
+  const sessionsTable =
+    schema.sessionsTable ??
+    (pgTable("session", {
+      sessionToken: text("sessionToken").primaryKey(),
+      userId: text("userId")
+        .notNull()
+        .references(() => usersTable.id, { onDelete: "cascade" }),
+      expires: timestamp("expires", { mode: "date" }).notNull(),
+    }) satisfies DefaultPostgresSessionsTable)
 
-export const postgresVerificationTokensTable = pgTable(
-  "verificationToken",
-  {
-    identifier: text("identifier").notNull(),
-    token: text("token").notNull(),
-    expires: timestamp("expires", { mode: "date" }).notNull(),
-  },
-  (table) => {
-    return {
-      compositePk: primaryKey({ columns: [table.identifier, table.token] }),
-    }
+  const verificationTokensTable =
+    schema.verificationTokensTable ??
+    (pgTable(
+      "verificationToken",
+      {
+        identifier: text("identifier").notNull(),
+        token: text("token").notNull(),
+        expires: timestamp("expires", { mode: "date" }).notNull(),
+      },
+      (table) => {
+        return {
+          compositePk: primaryKey({ columns: [table.identifier, table.token] }),
+        }
+      }
+    ) satisfies DefaultPostgresVerificationTokenTable)
+
+  return {
+    usersTable,
+    accountsTable,
+    sessionsTable,
+    verificationTokensTable,
   }
-) satisfies DefaultPostgresVerificationTokenTable
+}
 
 export function PostgresDrizzleAdapter(
   client: PgDatabase<QueryResultHKT, any>,
-  schema: DefaultPostgresSchema = {
-    usersTable: postgresUsersTable,
-    accountsTable: postgresAccountsTable,
-    sessionsTable: postgresSessionsTable,
-    verificationTokensTable: postgresVerificationTokensTable,
-  }
+  schema?: DefaultPostgresSchema
 ): Adapter {
   const { usersTable, accountsTable, sessionsTable, verificationTokensTable } =
-    schema
+    defineTables(schema)
 
   return {
     async createUser(data: AdapterUser) {
+      const { id, ...insertData } = data
       const hasDefaultId = getTableColumns(usersTable)["id"]["hasDefault"]
 
       return client
         .insert(usersTable)
-        .values(hasDefaultId ? data : { ...data, id: crypto.randomUUID() })
+        .values(
+          hasDefaultId ? insertData : { ...insertData, id: crypto.randomUUID() }
```

Comment:
could save some cycles and re-use the id coming from auth.js as the fallback?
```suggestion
          hasDefaultId ? insertData : { ...insertData, id }
```

Reply:
True, the `id` coming from auth.js will already be the result of a `crypto.randomUUID()` call. Can you do this update in all db types (postgres, mysql, sqlite)? :pray:

Reply:
yep, will update now